### ISSUE TITLE
Fix false positives with trailing comment after last pair in map

### ISSUE THREAD

**Issue by ghost** [stylelint/stylelint#4745]
> Clearly describe the bug

Hi everyone!
I have a very strange and ugly bug with Stylelint.
Take this piece of code:
```scss
$colors: (

primary: (
  base: #3557f0,
  dark: scale-color(#3557f0, $lightness: -13%), 
 transparent: rgba(#3557f0, .3), 
),

secondary: (
 base: #f09b2b,
 dark: scale-color(#f09b2b, $lightness: -13%), 
 transparent: rgba(#f09b2b, .3), 
),

accent: (
 base:#191919,
 light: scale-color(#191919, $lightness: 6%), 
 dark: scale-color(#191919, $lightness: -12%), 
 darker: scale-color(#191919, $lightness: -25%),
 transparent-light: rgba(scale-color(#191919, $lightness: 9%), .4), // rgba(46, 46, 46, 0.4)
),

red: (
 base: #f0372d,
 dark: scale-color(#f0372d, $lightness: -13%), 
 transparent: rgba(#f0372d, .3), 
),

green: (
 base:#37e62d,
 dark: scale-color(#37e62d, $lightness: -13%), 
 transparent: rgba(#37e62d, .3), 
),

black: (
 base: #000,
 bg: scale-color(#000, $lightness: 4%), 
),

white: (
  base: #fff,
  text: scale-color(#fff, $lightness: -9%),
  transparent-01: rgba(#fff, .1), 
  transparent-05: rgba(#fff, .5), 
  transparent-07: rgba(#fff, .7), 
),

brands: (
  dropbox: #0062ff,
  slack: #2eb67d,
  evernote: #4caf50,
  sketch: #f2bb22,
),

);
```
When I run stylelint it throws an error saying "Expected indentation of 2 spaces (indentation)" on the first map," Expected indentation of 4 spaces (indentation)" on the second map, "Expected indentation of 6 spaces (indentation)" on the third map and so on. Very strange, and when I try to run the auto fix the code becomes (prepare to suffer):

```scss

// Map containing site's colors
$colors: (

  primary: (
    base: #3557f0,
    dark: scale-color(#3557f0, $lightness: -13%), 
 transparent: rgba(#3557f0, .3), 
),

secondary: (
 base: #f09b2b,
 dar    k: scale-color      #f09b2b, $lightn      ss: -13%), 
 transparent: rgba(#f09b2b, .3), 
),

accent: (
 base:#191919,
 light: scale-color(#191919, $lightness: 6%      ), 
 dark:        scale-color(#19        919, $lightness: -12%), 
 darker: scale-color(#191919, $lightness: -25%), 
 transparent-light: rgba(scale-color(#191919, $lightness: 9%), .4), 
),

red: (
 base: #f0372d,
 dark: scale-color(#f0372d, $lightness: -13%), 
 transparent: rgba(#f0372d, .3), 
),

green: (        
 base:#          7e62d,
 dark: s          ale-color(#37e62d, $lightness: -13%), 
 transparent: rgba(#37e62d, .3), 
),

black: (
 base: #000,
 bg: scale-color(#00          0, $lightn            ss: 4%), 
),
            
white: (
  base: #fff,
  text: scale-color(#fff, $lightness: -9%), 
  transparent-01: rgba(#fff, .1), 
  transparent-05            : rgba(#ff              , .5), 
  tr              nsparent-07: rgba(#fff, .7), 
),


brands: (
  dropb              ox: #0062f                
  slack: #2                67d,
  evernote: #4caf50,
  sketch: #f2bb22,
),

)                                                                                                                      ;
```
The code is mess up.
> Which rule, if any, is the bug related to?

`"indentation": 2`

> What code is needed to reproduce the bug?

The code of the example above.

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "indentation": 2
  }
}
```

> Which version of stylelint are you using?

e.g. `9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Vs code extension

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, SCSS

> What did you expect to happen?

A correct two spaces indentation.

> What actually happened (e.g. what warnings or errors did you get)?

The code is completely messy up.

**Comment by Unknow-max** [comment]
Some screenshot of the errors:
![immagine](https://user-images.githubusercontent.com/59534318/81183516-8ca69e80-8faf-11ea-94ee-0b032d3feea2.png)
![immagine](https://user-images.githubusercontent.com/59534318/81183553-93cdac80-8faf-11ea-8c15-3de273b81106.png)
![immagine](https://user-images.githubusercontent.com/59534318/81183600-a1833200-8faf-11ea-944b-80d84fcacf9e.png)

**Comment by jeddy3** [comment]
@Unknow-max Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIAkBbAhgB3gAgBQB0A7IgS0MJgCccDDtsNKItsA3NAGwBpsj7HmOdt2wB6UdkgoUMQgBciASi5FS5KjT4MmLYSrrbBbTvuWrCigNwguIAM4BPeWgAeCcHbs3wEQgDMSAHN3YC18EEoAVw4YO3CcUIN6cLJYZzkSX3jsACYtAF8ifJB8oA) with:

```scss
$map: (

inner: (
  prop: val, 
  prop: val, // comment
),

inner: (
  prop: val,
  prop: val,
),

);
```

It seems the trailing single-line comment after the last pair in a set is causing the problem.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
